examples/tutorials/streamwriter_basic_tutorial.py

```diff
@@ -0,0 +1,658 @@
+    s.write_audio_chunk(0, WAVEFORM)
+
+######################################################################
+#
+
+Audio(path)
+
+######################################################################
+#
+# Now we look into each step in more detail.
+
+######################################################################
+#
+# Write destination
+# -----------------
+#
+# StreamWriter supports different types of write destinations
+#
+# 1. Local file
+# 2. File-like objects
+# 3. Streaming protocols (such as RTMP and UDP)
+# 4. Media devices (speakers and video players)†
+#
+# † Media devices will be explained in a separate tutorial.
+#
+
+######################################################################
+# Local files
+# ~~~~~~~~~~~
+#
+# StreamWriter supports saving media to local files.
+#
+#
+# .. code::
+#
+#    StreamWriter(dst="audio.wav")
+#
+#    StreamWriter(dst="audio.mp3")
+#
+# This works for still images and videos as well.
+#
+# .. code::
+#
+#    StreamWriter(dst="image.jpeg")
+#
+#    StreamWriter(dst="video.mpeg")
+#
+
+######################################################################
+# Streaming protocols
+# ~~~~~~~~~~~~~~~~~~~
+#
+# You can stream the media with streaming protocols
+#
+# .. code::
+#
+#    # Real-Time Messaging Protocol
+#    StreamWriter(dst="rtmp://localhost:1234/live/app", format="flv")
+#
+#    # UDP
+#    StreamWriter(dst="udp://localhost:48550", format="mpegts")
+#
+
+######################################################################
+# File-like objects
+# ~~~~~~~~~~~~~~~~~
+#
+# You can also pass a file-lie object. A file-like object must implement
+# ``write`` method conforming to :py:attr:`io.RawIOBase.write`.
+#
+# .. code::
+#
+#    # Open the local file as fileobj
+#    with open("audio.wav", "wb") as dst:
+#        StreamWriter(dst=dst)
+#
+# .. code::
+#
+#    # In-memory encoding
+#    buffer = io.BytesIO()
+#    StreamWriter(dst=buffer)
+#
+
+######################################################################
+#
+# Configuring output streams
+# --------------------------
+#
+# Once the destination is specified, the next step is to configure the streams.
+# For typical audio and still image cases, only one stream is required,
+# but for video with audio, at least two streams (one for audio and the other
+# for video) need to be configured.
+#
+
+######################################################################
+# Audio Stream
+# ~~~~~~~~~~~~
+#
+# An audio stream can be added with
+# :py:meth:`~torchaudio.io.StreamWriter.add_audio_stream` method.
+#
+# For writing regular audio files, providing `sample_rate` and `num_channels`
+# is sufficient
+#
+# .. code::
+#
+#    s = StreamWriter("audio.wav")
+#    s.add_audio_stream(sample_rate=8000, num_channels=2)
+#
+# If the audio waveform to be written is of data type other than ``torch.float32``,
+# then, you can specify it with ``format`` option.
+#
+# .. code::
+#
+#    # Audio data passed to streamer must be torch.int16
```

Comment:
must be `torch.int16` or `torch.float32`? do you mean it is a note for describing the real format of the audio data?

Reply:
I will revisit this for easier understanding.

Reply:
@nateanl I revised the explanation of formats. How does it sound now?